_posts/2024-03-19-scalapay.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+title: "[CVE-2024-28393] Improper neutralization of SQL parameter in Scalapay module for PrestaShop"
+categories: modules
+author:
+- Touchweb.fr
+- 202 ecommerce.com
+meta: "CVE,PrestaShop,scalapay"
+severity: "critical (9.8)"
+---
+
+In the module "Scalapay" (scalapay) up to version 1.2.41 from Scalapay for PrestaShop, a guest can perform SQL injection in affected versions.
+
+## Summary
+
+* **CVE ID**: [CVE-2024-28393](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-28393)
+* **Published at**: 2024-03-19
+* **Advisory source**: Friends-Of-Presta.org
+* **Platform**: PrestaShop
+* **Product**: scalapay
+* **Impacted release**: <= 1.2.41 (1.2.42 fixed the vulnerability)
+* **Product author**: Scalapay
+* **Weakness**: [CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+* **Severity**: critical (9.8)
+
+## Description
+
+The method `ScalapayReturnModuleFrontController::postProcess()` has sensitive SQL calls that can be executed with a trivial http call and exploited to forge a SQL injection.
+
+This exploit uses a PrestaShop front controller and most attackers can conceal the module controller's path during the exploit, so you will never know within your conventional frontend logs that it exploits this vulnerability. **You will only see "POST /" inside your conventional frontend logs.** Activating the AuditEngine of mod_security (or similar) is the only way to get data to confirm this exploit.
+
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: none
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: [CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H)
+
+## Possible malicious usage
+
+* Obtain admin access
+* Remove data from the associated PrestaShop
+* Copy/paste data from sensitive tables to FRONT to expose tokens and unlock admin's ajax scripts
+* Rewrite SMTP settings to hijack emails
+
+## Patch from 1.2.41
+
+```diff
+--- 1.2.41/modules/scalapay/controllers/front/return.php
++++ 1.2.42/modules/scalapay/controllers/front/return.php
+...
+            if ((!isset($cart) or !$cart->id) && $cart_id_return != '') {
+                // get customer id from cart table
+-               $query_scalapay_get = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS("SELECT id_customer FROM " . _DB_PREFIX_ . "cart WHERE id_cart='" . $cart_id_return . "'");
++               $query_scalapay_get = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS("SELECT id_customer FROM " . _DB_PREFIX_ . "cart WHERE id_cart='" . pSQL($cart_id_return) . "'");
```

Comment:
it is an int ? => (int) ?

Reply:
J'ai laissé le fix de l'auteur - il répond au besoin mais vous savez tous que je suis d'accord avec vous.